lib/compilation_engine.rb

```diff
@@ -237,8 +285,12 @@ def compile_term
       output_token # )
 
     else
+      name = @tokenizer.identifier
       output_token # int / str / keyword / identifier / start of a subroutine call
-
+      
+      if @tokenizer.token_type == :IDENTIFIER
```

Comment:
There could be other tokens here, but we only want to output information if the term is an identifier